pwa-devdocs/_drafts/state-management/index.md

```diff
@@ -0,0 +1,188 @@
+---
+title: State management
+---
+
+<!-- TODO: publish this under the technologies/basic-concepts directory -->
+
+## What is state management
+
+An application, such as a PWA storefront, uses state data to render dynamic content to the user.
+State management describes the way the application handles changing state data as a result of user interactions.
+
+Interactions, such as clicking on a button or loading the page, modify the state and update the appearance or behavior of the application.
+
+For example, a shopper clicks on a button to add an item to the shopping cart.
+The application needs a way to add that item to the shopping cart while the shopper continues to browse the application.
+It also needs to update the visual components that use shopping cart data to reflect the new state.
+
+## Local versus global state
+
+Local and global are the two different types of state a component can depend on.
+
+Local state data is any data scoped within a component or its children components.
+This type of data is not shared with a component's parent or peer data.
+Otherwise, that data should be [lifted][].
+
+For example, form data is never used outside of that components, so
```

Comment:
A better example may be something like whether a button is disabled after clicking it. For example:

```js
const [isDisabled, setIsDisabled] = useState(false);

const handleClick= useCallback(() => setIsDisabled(true),[])

return <button onClick={handleClick} disabled={isDisabled}>Click me!</button>;
```